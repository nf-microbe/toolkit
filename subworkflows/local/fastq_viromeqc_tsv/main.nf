include { VIROMEQC_INSTALL  } from '../../../modules/nf-core/viromeqc/install/main'
include { VIROMEQC_VIROMEQC } from '../../../modules/nf-core/viromeqc/viromeqc/main'

workflow FASTQ_VIROMEQC_TSV {

    take:
    fastq_gz    // [ [ meta ]   , [ reads_1.fastq.gz, reads_2.fastq.gz ]
    viromeqc_db // [ viromeqc_db ]

    main:
    ch_versions = Channel.empty()

    // if viromeqc_db exists, skip VIROMEQC_INSTALL
    if (viromeqc_db){
        ch_genomad_db = viromeqc_db
    } else {
        //
        // MODULE: download viromeQC database
        //
        VIROMEQC_INSTALL(
        )
        ch_viromeqc_db  = VIROMEQC_INSTALL.out.viromeqc_db
        ch_versions     = ch_versions.mix(VIROMEQC_INSTALL.out.versions)
    }

    //
    // MODULE: Estimate viral enrichment
    //
    VIROMEQC_VIROMEQC(
        fastq_gz,
        ch_viromeqc_db
    )
    ch_versions         = ch_versions.mix(VIROMEQC_VIROMEQC.out.versions)

    emit:
    enrichment_tsv  = VIROMEQC_VIROMEQC.out.tsv // channel: [ [ meta ], TSV ]
    versions        = ch_versions                      // channel: [ versions.yml ]
}

