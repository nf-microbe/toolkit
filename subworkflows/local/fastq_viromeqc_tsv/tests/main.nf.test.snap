{
    "sarscov2 - fastq - pe - no db": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test1",
                            "single_end": false
                        },
                        "test1.viromeqc.tsv:md5,87fe55cead9456793cab8c88cec19745"
                    ],
                    [
                        {
                            "id": "test2",
                            "single_end": false
                        },
                        "test2.viromeqc.tsv:md5,9b8938d58b2d61b9eea191d6e8585151"
                    ]
                ],
                "1": [
                    "versions.yml:md5,0b1b887ab740e7c151ce5620bca52444",
                    "versions.yml:md5,0b1b887ab740e7c151ce5620bca52444",
                    "versions.yml:md5,790fa966ff74fe8ff73caa4333418c0d"
                ],
                "enrichment_tsv": [
                    [
                        {
                            "id": "test1",
                            "single_end": false
                        },
                        "test1.viromeqc.tsv:md5,87fe55cead9456793cab8c88cec19745"
                    ],
                    [
                        {
                            "id": "test2",
                            "single_end": false
                        },
                        "test2.viromeqc.tsv:md5,9b8938d58b2d61b9eea191d6e8585151"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,0b1b887ab740e7c151ce5620bca52444",
                    "versions.yml:md5,0b1b887ab740e7c151ce5620bca52444",
                    "versions.yml:md5,790fa966ff74fe8ff73caa4333418c0d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-23T15:19:10.005049815"
    }
}